models/DataObject/ClassDefinition/Data/Input.php

```diff
@@ -327,7 +327,20 @@ public function getQueryColumnType()
     public function checkValidity($data, $omitMandatoryCheck = false, $params = [])
     {
         if (!$omitMandatoryCheck && $this->getRegex() && strlen($data) > 0) {
-            if (!preg_match('#' . $this->getRegex() . '#' . implode('', $this->getRegexFlags()), $data)) {
+            $throwException = false;
+            if(in_array('g', $this->getRegexFlags())){
+                $flags = str_replace('g', '', implode('', $this->getRegexFlags()));
+                if (!preg_match_all('#' . $this->getRegex() . '#' . $flags, $data)) {
+                    $throwException = true;
+                }
+            }
+            else{
+                if (!preg_match('#' . $this->getRegex() . '#' . implode('', $this->getRegexFlags()), $data)) {
+                    $throwException = true;
+                }
+            }
```

Comment:
This should be enough, right?

```suggestion
            $flags = str_replace('g', '', implode('', $this->getRegexFlags()));
            if (!preg_match_all('#' . $this->getRegex() . '#' . $flags, $data)) {
                $throwException = true;
            }
```